pkg/apis/resource/types.go

```diff
@@ -139,6 +139,53 @@ type ResourceSliceSpec struct {
 	// +optional
 	// +listType=atomic
 	Devices []Device
+
+	// PerDeviceNodeSelection defines whether the access from nodes to
+	// resources in the pool is set on the ResourceSlice level or on each
+	// device. If it is set to true, every device defined the ResourceSlice
+	// must specify this individually.
+	//
+	// Exactly one of NodeName, NodeSelector, AllNodes, and PerDeviceNodeSelection
+	// must be set.
+	//
+	// +optional
+	// +oneOf=NodeSelection
+	PerDeviceNodeSelection bool
```

Comment:
All net-new fields in non-alpha structs need to be nilable (maps, slices, or pointers).

Reply:
Updated